helper/pkcs7/ber.go

```diff
@@ -149,14 +149,14 @@ func readObject(ber []byte, offset int) (asn1Object, int, error) {
 		for ber[offset] >= 0x80 {
 			tag = tag*128 + ber[offset] - 0x80
 			offset++
-			if offset > berLen {
+			if offset >= berLen {
```

Comment:
nit: curious if it is worth adding a length check test for the panic fix case to [ber_test.go](https://github.com/hashicorp/vault/blob/main/helper/pkcs7/ber_test.go)?

Reply:
I had to update couple tests so we are getting some coverage for this change. I do think it would be good to add a test for the specific panic case that was found but until we get more info I don't know how to repro that.

Reply:
A fuzz test could be useful to help discover the parameters that causes a panic:
```go
func FuzzReadObject(f *testing.F) {
	f.Fuzz(func(t *testing.T, ber []byte, offset int) {
		if offset < 0 {
			return
		}
		_, _, err := readObject(ber, offset)
		if err != nil {
			t.Log(ber, offset)
		}
	})
}
```

Reply:
ah, of course! Good thinking :)

Reply:
done in https://github.com/hashicorp/vault/pull/24891/commits/52775da5ce8d1f3808c9c81b0621962a99ea841a

Reply:
I just used the inputs from `TestBer2Der_Negatives` as the seed corpus and it found the panic very quickly:

```go
func FuzzReadObject(f *testing.F) {
       testcases := []struct {
               input  []byte
               offset int
       }{
               {[]byte{0x30, 0x85}, 0},
               {[]byte{0x30, 0x84, 0x80, 0x0, 0x0, 0x0}, 0},
               {[]byte{0x30, 0x82, 0x0, 0x1}, 0},
               {[]byte{0x30, 0x80, 0x1, 0x2, 0x1, 0x2}, 0},
               {[]byte{0x30, 0x80, 0x1, 0x2}, 0},
               {[]byte{0x30, 0x03, 0x01, 0x02}, 0},
               {[]byte{0x30}, 0},
       }
       for _, tc := range testcases {
               f.Add(tc.input, tc.offset) // Use f.Add to provide a seed corpus
       }
       f.Fuzz(func(t *testing.T, ber []byte, offset int) {
               if offset < 0 {
                       return
               }
               _, _, err := readObject(ber, offset)
               if err != nil {
                       t.Log(ber, offset)
               }
       })
}
```

Reply:
I think that this would be great to add to the test file!